### PR TITLE
DOC-804: Update inverted indexes to GIN indexes

### PR DESCRIPTION
**Pull request by nickvigilante**
Fixes DOC-804 and DOC-1528:

- Rename references of inverted indexes to GIN indexes
- Update SEO

### CODE REVIEW

**Comment by nickvigilante** [comment]
I intend to push these changes down to 21.1 and 20.2. A few questions before I do:

1. Should we change the actual name of the page from inverted-indexes.md to gin-indexes.md?
2. Are the changes I've made sufficient at least for 21.2?

**Review by awoods187 — APPROVED** [review]
> Reviewed 1 of 10 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/12826)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @vy-ton)

**Comment by awoods187** [comment]
Can we check that we have an alias in docs search as well as for SEO? If someone knew the title "inverted index" we want them to be able to easily find "GIN"

**Review by vy-ton — APPROVED** [review]

**Comment by nickvigilante** [comment]
@awoods187 I think I found a bug with either Netlify or Algolia or some other component related to search where it's not actually pulling in the changes I made. If you search for the word GIN, you only get 3 results. Therefore, unless that's fixed, there will be no way to check if SEO is working correctly. Although I did keep the name of the page the same as the old page so there's a chance that if you search for inverted indexes it'll still return the same page and results side by side. Gonna talk to Lauren about how to address.

**Comment by nickvigilante** [comment]
@awoods187 I confirmed with Lauren that we configured Algolia to hit only the PROD docs site because of cost, so I won't know about SEO until we publish this. I will check once it does get published to ensure it works, and if it doesn't, I'll log another JIRA/PR for follow-up. Other than that, does this look OK to you?

**Comment by awoods187** [comment]
SGTM! Nice work!

**Review by gemma-shay — APPROVED** [review]
Should `inverted-indexes.html` be renamed? I assume it's fine, just checking. Caught a few "an GIN" instances but LGTM otherwise!

File: releases/v21.1.0-alpha.1.md [hunk 1]
```diff
@@ -254,15 +255,15 @@ $ docker pull cockroachdb/cockroach-unstable:v21.1.0-alpha.1
...
-- The optimizer may now plan an inverted join if two tables are joined on `JSONB` or `ARRAY` columns using a contains predicate (e.g., `WHERE a @> b`), and the first column has an inverted index. The inverted join will be chosen if the optimizer expects it to be more efficient than any alternative plan. For queries in which the only alternative is a cartesian product followed by a filter, the inverted join will likely result in a performance improvement. [#55679][#55679]
+- The optimizer may now plan an inverted join if two tables are joined on `JSONB` or `ARRAY` columns using a contains predicate (e.g., `WHERE a @> b`), and the first column has an GIN index. The inverted join will be chosen if the optimizer expects it to be more efficient than any alternative plan. For queries in which the only alternative is a cartesian product followed by a filter, the inverted join will likely result in a performance improvement. [#55679][#55679]
```
[the remark this answers is not in the document]
Comment (nickvigilante) [reply releases/v21.1.0-alpha.1.md]: Done

File: v21.1/inverted-indexes.md [hunk 2]
```diff
@@ -138,14 +139,14 @@ If you require comparisons using [`<`](functions-and-operators.html#supported-op
...
+GIN indexes on [`ARRAY`](array.html) columns support the following comparison operators:
 
 - "is contained by": [`<@`](functions-and-operators.html#supported-operations)
 - "contains": [`@>`](functions-and-operators.html#supported-operations)
 
-## Partial inverted indexes
+## Partial GIN indexes
 
-<span class="version-tag">New in v21.1:</span> You can create a [partial](partial-indexes.html) inverted index, an inverted index on a subset of `JSON`, `ARRAY`, or geospatial container column data. Just like partial indexes that use non-container data types, create a partial inverted index by including a clause that evaluates to true on a boolean predicate, like a `WHERE` clause.
+<span class="version-tag">New in v21.1:</span> You can create a [partial](partial-indexes.html) GIN index, an GIN index on a subset of `JSON`, `ARRAY`, or geospatial container column data. Just like partial indexes that use non-container data types, create a partial GIN index by including a clause that evaluates to true on a boolean predicate, like a `WHERE` clause.
```
[the remark this answers is not in the document]
Comment (nickvigilante) [reply v21.1/inverted-indexes.md]: Done

File: v21.2/inverted-indexes.md [hunk 3]
```diff
@@ -158,15 +159,15 @@ CREATE TABLE test (
...
 
 {% include {{page.version.version}}/sql/indexes-regional-by-row.md %}
 
 For an example that uses unique indexes but applies to all indexes on `REGIONAL BY ROW` tables, see [Add a unique index to a `REGIONAL BY ROW` table](add-constraint.html#add-a-unique-index-to-a-regional-by-row-table).
 
-## Multi-column inverted indexes
+## Multi-column GIN indexes
 
-You can create an inverted index with multiple columns. The last indexed column must be one of the inverted types such as `JSON`, `ARRAY`, `GEOMETRY`, and `GEOGRAPHY`. All preceding columns must have types that are indexable. These indexes may be used for queries that constrain all index columns.
+You can create an GIN index with multiple columns. The last indexed column must be one of the inverted types such as `JSON`, `ARRAY`, `GEOMETRY`, and `GEOGRAPHY`. All preceding columns must have types that are indexable. These indexes may be used for queries that constrain all index columns.
```
[the remark this answers is not in the document]
Comment (nickvigilante) [reply v21.2/inverted-indexes.md]: Done

File: v21.2/inverted-indexes.md [hunk 4]
```diff
@@ -298,7 +299,7 @@ Insert a few rows of data:
 (3 rows)
 ~~~
 
-Now, let’s add an inverted index to the table and run a query that filters on the `ARRAY`:
+Now, let’s add an GIN index to the table and run a query that filters on the `ARRAY`:
```
[the remark this answers is not in the document]
Comment (nickvigilante) [reply v21.2/inverted-indexes.md]: Done

**Comment by nickvigilante** [comment]
@gemma-shay I think for SEO (in the event my keyword changes do not work), it would be good to still have the old page name to increase searchability. However, I'll add this as a discussion point during my schema weekly.

**Pull request merged**